modules/registry-configuring-pullsecrets-service-accounts.adoc

```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+//
+// * openshift_images/managing_images/using-image-pull-secrets.adoc
+// * authentication/understanding-and-creating-service-accounts.adoc
+
+
+:_content-type: CONCEPT
+[id="registry-configuring-pullsecrets-service-accounts_{context}"]
+= Manually configuring service account pull secrets for the integrated image registry
+
+When a service account is created, it is automatically provisioned with pull secrets for the integrated image registry. The provisioning of pull secrets happens quickly, but is not immediate.
+
+[WARNING]
+====
+If a pod is created before the service account is provisioned with the pull secret, it cannot pull images from the image registry even after the service account pull secret is  provisioned.
+====
+
+If you need to create workloads or pods that pull images from the integrated image registry immediately, you can manually configure service account pull secrets for the integrated image registry.
+
+.Procedure
+
+* To create a `./config.json` file with a pull secret for the service account, enter the following command: 
++
+[source,terminal]
+----
+$ oc registry login -n <NAMESPACE> -z <SERVICEACCOUNT> --to=./config.json
```

Comment:
then create a pull secret:

```
oc create secret generic <PULL_SECRET_NAME> \
    --from-file=.dockerconfigjson=./config.json \
    --type=kubernetes.io/dockerconfigjson
```

and link it to the service account:

```
oc secrets link <SERVICEACCOUNT> <PULL_SECRET_NAME> --for=pull
```

Reply:
@xiuwang ^^ can you check/confirm it?

Reply:
@dmage @bmcelvee Found a bug when use serviceaccount to login image registry 
https://bugzilla.redhat.com/show_bug.cgi?id=2093797 
Could use the workaround
`oc login -u <username> -p <password> https://<cluster-api>:6443`
`oc registry login --to=./config.json`

Reply:
@xiuwang this will put your personal token into ./config.json, but personal tokens have limited life time.

Reply:
Ok, then we are better to wait the bug fixed, it has been set to blocker+ flag.

Reply:
@dmage @bmcelvee  https://github.com/openshift/oc/pull/1166 will remove the service account, we could use `--auth-basic` to get the image registry auth

Reply:
@soltysh can you guide how to create a pull secret for the service account in 4.9+?

Reply:
The auth-basic options usage could be:
```
token=$(oc create token  <SERVICEACCOUNT>)
oc registry login --auth-basic=anyuser:$token --to=./config.json
```

Reply:
I guess we have to have different steps for 4.10 and 4.11?.. That's not the best UX if 4.10 steps are not valid in 4.11.